packages/spark-core/src/interceptors/spark-tracking-id.js

```diff
@@ -7,51 +7,28 @@
 import {Interceptor} from '@ciscospark/http-core';
 import uuid from 'uuid';
 
+const sequenceNumbers = new WeakMap();
+
 /**
  * @class
  */
 export default class SparkTrackingIdInterceptor extends Interceptor {
   /**
-   * @constructor
-   * @param {ProxySpark} spark
-   * @param {Object} options
-   * @returns {SparkTrackingIdInterceptor}
+   * Sequence number; increments on access
+   * @type {Number}
    */
-  constructor(spark, options) {
-    super({spark});
-
-    if (!options || !options.prefix) {
-      throw new Error(`\`options.prefix\` is required. Did you forget to specify it at config.trackingIdPrefix?`);
-    }
-
-    let sequence = 0;
-    const base = options.base || uuid.v4();
-    const prefix = options.prefix || `spark-js-sdk`;
-
-    Object.defineProperties(this, {
-      base: {
-        writable: false,
-        value: base
-      },
-      prefix: {
-        writable: false,
-        value: prefix
-      },
-      sequence: {
-        get: function get() {
-          sequence += 1;
-          return sequence;
-        }
-      }
-    });
+  get sequence() {
+    let sq = sequenceNumbers.get(this) || 0;
+    sq += 1;
```

Comment:
their's an eslint rule set to discourage ++ and -- since they're not intuitive to newer developers